Readds OFD firing at away sites

RustingWithYou opened the pull request:
🆑 RustingWithYou
rscadd: The OFD can now fire at away sites again.
rscadd: The OFD now has strength and range settings on its console again.
/🆑 

With @SierraKomodo's teleporter rework approaching and making it more difficult to beam supplies to explorers, I figured that the ability to launch droppods should return, and I figured that it would be inconsistent to have the OFD be capable of launching crates to an away site but nothing else. Also, truth be told, the arguments for its removal seemed fairly weak to me to begin with, so I figured I'd open discussion of its return to the community in general.

RustingWithYou commented:
there are a few broken things here, closing until i can fix em

RustingWithYou commented:
I have fixed things, thank you @SealCure for telling me how to do code less bad.

Alex6511 commented on code/modules/overmap/disperser/disperser_charge.dm:
```diff
@@ -6,30 +6,59 @@
 	var/chargedesc
 
+/obj/structure/ship_munition/disperser_charge/proc/fire(turf/target, strength, range)
+	CRASH("OFD charge firing logic not set!")
```
What's the purpose of this?

RustingWithYou replied:
I'm not entirely sure as most of this code is just the old OFD/BSA code updated for overmap changes. If I had to guess I'd say it's to avoid the base type of OFD charge being used in the unlikely event that it actually winds up being spawned somehow.

Alex6511 commented on code/modules/overmap/disperser/disperser_charge.dm:
```diff
@@ -6,30 +6,59 @@
+		disperserf.lifetime = strength * 20
+
+/obj/effect/fake_fire/bluespace
+	name = "bluespace fire"
```
What is the purpose of this blue space fire?

RustingWithYou replied:
Fake fire is used to prevent fucking atmos to high hell, I believe. As for why it's blue, I presume it's because the original creator thought it looked cooler.

Alex6511 replied:
Messing with atmos on planets in any appreciable manner should be basically impossible.

RustingWithYou replied:
I presume the creator originally wanted to avoid igniting flammable planets with incendiary charges.

Shadowtail117 replied:
Many planets have ignitable atmospheres such that a single spark can set the whole thing on fire and leave it a 400000 degree 10000kpa death zone.

Alex6511 commented on code/modules/overmap/disperser/disperser_charge.dm:
```diff
@@ -6,30 +6,59 @@
+		if(prob(strength * 100 / 6)) //6 instead of 5 so there are always leftovers
+			M.GetDrilled(TRUE) //no artifacts survive this
+	for(var/mob/living/L in victims)
+		to_chat(L, "<span class='alert'>You feel an incredible power trying to pry you apart... But it decides to spare you.</span>")
```
Are the mining charges sentient? This seems like an odd choice.

Alex6511 commented on code/modules/overmap/disperser/disperser_fire.dm:
```diff
@@ -76,23 +76,46 @@
 	var/list/candidates = list()
 
-	for(var/obj/effect/overmap/event/O in get_step(linked, overmapdir))
+//Prioritize events. Thus you can hide in meteor showers in exchange for protection from the disperser.
```
This comment does not appear to be spaced correctly.

Alex6511 commented on code/modules/overmap/disperser/disperser_fire.dm:
```diff
@@ -101,6 +124,28 @@
+	var/area/finalarea = pick(targetareas)
+	var/turf/targetturf = pick_area_turf(finalarea.type, list(/proc/is_not_space_turf))
+
+	log_and_message_admins("Aforementioned disperser beam hit sector at [get_area(targetturf)].", location=targetturf)
```
This log message shouldn't refer to previous log messages and should stand on its own. Doing this is bad because if you got disjointed logs it might be difficult to link the two together.

Imagine a situation in which multiple events could happen at the same time, the "Aforementioned" beam could be several log messages above.

Alex6511 commented on code/modules/overmap/disperser/disperser_fire.dm:
```diff
@@ -101,6 +124,28 @@
+		charge.forceMove(targetturf)
+		//The disperser is not a taxi
+		for(var/mob/living/L in charge)
+			to_chat(L, SPAN_DANGER("As you pass through space you suddenly ram into the fourth wall."))
```
The person is killed during this are they not? If so I don't think this is needed. Alternatively, a description of them being ripped apart or something.

Alex6511 commented:
Many of my comments still stand.

Alex6511 reviewed (changes requested):
Many of my comments still stand.

Spookerton commented:
Leaving this open for the moment, but I don't like it as-is.

RustingWithYou committed:
readds OFD firing at away sites


tweaks to disperser console


fix to disperser charge


works around refactored overmap objs


minor tweaks to charge firing messages/admin notes

SierraKomodo commented on code/modules/overmap/disperser/disperser_charge.dm:
```diff
@@ -6,30 +6,59 @@
 	var/chargedesc
 
+/obj/structure/ship_munition/disperser_charge/proc/fire(turf/target, strength, range)
+	CRASH("OFD charge firing logic not set!")
```
It'd be preferred if the crash message was more descriptive of why it's actually crashing. 'OFD charge firing logic' implies there'd be some var or configuration that would cause this crash when really it just always happens when this proc is called on this specific item.

SierraKomodo commented on code/modules/overmap/disperser/disperser_charge.dm:
```diff
@@ -6,30 +6,59 @@
+		if(prob(strength * 100 / 6)) //6 instead of 5 so there are always leftovers
+			M.GetDrilled(TRUE) //no artifacts survive this
+	for(var/mob/living/L in victims)
+		to_chat(L, "<span class='alert'>You feel an incredible force ripping and tearing at you.</span>")
```
Span classes - Also, 'danger' is more appropriate for 'You're about to die'

```suggestion
		to_chat(L, SPAN_DANGER("You feel an incredible force ripping and tearing at you."))
```

SierraKomodo commented on code/modules/overmap/disperser/disperser_console.dm:
```diff
@@ -108,13 +110,19 @@ obj/machinery/computer/ship/disperser/proc/is_valid_setup()
 	if(B)
 		return B
 
+	var/obj/structure/closet/C = locate() in get_turf(back)
+	return C
```
```suggestion
	var/obj/structure/closet/C = locate() in get_turf(back)
	if (C)
		return C
```

SierraKomodo reviewed (dismissed):
Not an exhaustive review, just items I saw skimming over the changes.

RustingWithYou committed:
Update code/modules/overmap/disperser/disperser_charge.dm

Co-authored-by: Sierra Brown (SierraKomodo) <SierraKomodo@users.noreply.github.com>

SierraKomodo reviewed (commented):
@Spookerton For the most part it looks fine on my end. Nothing that's obviously broken, at least. Thoughts?

Spookerton reviewed (approved):
Sierra's welcome to merge this -- I'm just sad that the opportunity to do something better than randomly located explosions wasn't taken.
